### ISSUE TITLE
Crash while generating stack trace

### ISSUE THREAD

**Issue by digeff** [evanw/node-source-map-support#131]
While trying to translate a stack trace my project hits this case:

    // Fix position in Node where some (internal) code is prepended.
    // See https://github.com/evanw/node-source-map-support/issues/36
    if (line === 1 && !isInBrowser() && !frame.isEval()) {
      column -= 62;
    }

Which doesn't apply to my code, because it doesn't has the header mentioned in that issue... It just has code on the first line, but it has less than 62 columns so it crashes later when the column is negative.

Apart from that, if translating the stack trace to typescript fails I think you should just print the original javascript stacktrace, not throw an error.

**Comment by julien-f** [comment]
What doesn't it apply to your code, you which context this code is executed?

**Comment by digeff** [comment]
I execute my code in node, on line 1 and its not an eval, so it hits that condition but my code only has 50 columns long, so that line makes my program crash.

**Comment by digeff** [comment]
This particular file is .JS code, so it doesn't even have a .map file even.

**Comment by julien-f** [comment]
Node automatically wraps the code, therefore the first line is always shifted by 62 chars.

What version of Node are you using?

**Comment by digeff** [comment]
One of the latest. The issue though is probably that I'm running this code with vm.runScript or something like that, I'm not requiring it, so node is not adding that prefix.

**Comment by julien-f** [comment]
You mean you are reading the code yourself and run it through vm?

**Comment by digeff** [comment]
Yes

You mean you are reading the code yourself and run it through vm?

-

**Comment by julien-f** [comment]
I am not sure what can be done to fix this issue properly.

We could at least make for line sorter than 62 chars…

**Comment by digeff** [comment]
Yes, put a check so column doesn't go into negative numbers or zero

**Comment by digeff** [comment]
You could also check if the file starts with
(function (exports, blah blah blah)) // the 62 character string
and only remov ethe 62 characters if the file starts with that prefix

**Comment by julien-f** [comment]
This prefix is only present in memory, I don't think we have access to it…

**Comment by digeff** [comment]
ok, what about updating the code to if (column >= 63) { column -= 62 } // at least?

**Comment by julien-f** [comment]
Yep, could be done, would be nice to have tests though.
